lib/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 if(NOT CONFIG_NATIVE_APPLICATION)
 add_subdirectory(libc)
 endif()
-add_subdirectory_ifdef(CONFIG_POSIX_API            posix)
+add_subdirectory(posix)
```

Comment:
If you include this always, you should have all its individual c files gated with something else

Reply:
> If you include this always, you should have all its individual c files gated with something else

Probably yes, and eventually yes. But what happens when all those gates are enabled? Random build failures from native_posix. And that's the problem and means to address it which we discuss.

Reply:
My comment was related to a) code side & b) if you include them without any Kconfig option, then you need to be careful that today posix_cheats only does the renaming if POSIX_API is set

Reply:
> then you need to be careful that today posix_cheats only does the renaming if POSIX_API is set

One way to be careful is again to never include posix_cheats when anything POSIX subsys related is used. As posix_cheats is mandatory for ARCH_POSIX, that leads to obvious conclusions.